### PR TITLE
Admin homepage requestgrouptable

### PR DESCRIPTION
**Pull request by guy-ro**
- new resolver for nextRecipient for requestGroups and requestTypes (returns Client that is next in line (the target of the last-to-be-added Request that is not fulfilled or deleted)
- new resolver for hasAnyRequests for requestGroups to determine whether any Requests are in any requestType in that requestGroup
- added component for requestGroupTable (width is 100% of parent width so put it inside a properly sized div when using) (component expects an array of requestGroups with all the necessary properties)
- modified some of the common table styling provided by @Anand-Issac 
- modified Tag styling to be used with more versatility

image
image
image
image

### CODE REVIEW

**Comment by judyzhong27** [comment]
Notes on UI:

- Bottom gray line underneath the headers of the table (Title, Item Types, Open Requests) seems thicker/darker than the line above
- The lines that are not around the headers and are between individual rows should not extend to either end of the screen. It's slightly shorter than the header lines. See image
Screen Shot 2021-04-01 at 1 52 53 PM
- The table contents should be equidistant on both sides of the screen. Right now it seems too right-skewed with excess padding on the left and the image is too close to the edge of the right. Could have something to do with the max width of each column, but it works on the design for 1440x900 in Figma. Not sure what screen size this screenshot is for.
- Appears to have too much padding between the Open Requests column and Item Types column

**Comment by guy-ro** [comment]
@judyzhong27 

Made requested modifications:

image
image

**Comment by judyzhong27** [comment]
@guy-ro 

Looks great! Quick note with this screenshot:

Screen Shot 2021-04-01 at 10 22 17 PM

Not sure why there is a space between the lines and the edge here? Maybe the screenshot is off but lines should be to the edge of screen. Everything else is good!

**Comment by guy-ro** [comment]
> @guy-ro
> 
> Looks great! Quick note with this screenshot:
> 
> Screen Shot 2021-04-01 at 10 22 17 PM
> 
> Not sure why there is a space between the lines and the edge here? Maybe the screenshot is off but lines should be to the edge of screen. Everything else is good!

We don't have the page where the table will live setup yet. For testing, I simply put it on a page where it doesn't belong. @jlight99, make sure to make the parent div of the table has width = 100vw

Thanks for the review Judy!

File: client/src/components/molecules/RequestGroupTable.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                        }
+                        if (!g1 || !g1.name) {
+                            return 1
+                        }
+                        if (!g2 || !g2.name) {
+                            return -1
+                        }
+                        return g1.name < g2.name ? -1 : 1})
+                    .map((requestGroup : RequestGroup) => 
+                        <>
```
Comment (meganniu) [inline client/src/components/molecules/RequestGroupTable.tsx:46]: LOL ive never seen an element like this `<>`. What is it?

Comment (meganniu) [inline client/src/components/molecules/RequestGroupTable.tsx:46]: LOL ive never seen an element like this `<></>`. What is it?

**Review by meganniu — APPROVED** [review]
LGTM!!! lol ty for all ur styling pains

File: client/src/components/molecules/RequestGroupTable.tsx (hunk 1 above)
Comment (guy-ro) [reply client/src/components/molecules/RequestGroupTable.tsx:46]: It's syntactic sugar for a React fragment. Just a simpler way to write it.

**Review by jlight99 — APPROVED** [review]
looks beautiful!

**Pull request merged**